src/main/java/cuchaz/enigma/source/procyon/index/TokenFactory.java

```diff
@@ -17,6 +17,11 @@ public static Token createToken(SourceIndex index, AstNode node) {
         String name = node instanceof Identifier ? ((Identifier) node).getName() : "";
         Region region = node.getRegion();
 
+        if (region.getBeginLine() == 0) {
+            System.err.println("Got bad region from Procyon for node " + node);
```

Comment:
Most of the time it is like java output visitor's bug that it doesn't update the region of the token when it does java output

Reply:
Yes, this is probably the cause of the bug. But it's in Procyon's code, not Enigma's code, so all we can do from Enigma is add a check.